client/allocrunner/alloc_runner.go

```diff
@@ -1348,6 +1348,9 @@ func (ar *allocRunner) Reconnect(update *structs.Allocation) (err error) {
 	// Build the client allocation
 	alloc := ar.clientAlloc(states)
 
+	// Set the reconnect modify index so that the scheduler can track that the reconnect has not been processed.
+	alloc.ReconnectModifyIndex = ar.Alloc().ReconnectModifyIndex
```

Comment:
I don't think I understand what ever sets `ReconnectModifyIndex` to a non-zero value outside of tests. The approach seems right here but I can't figure out the source of the index.

Reply:
I'm not entirely sure what happened with that refactoring. It was getting set in `setIndexex` and then I moved it over here.  I just updated to `alloc.ReconnectModifyIndex = ar.Alloc().AllocModifyIndex`.